lib/active_model/serializer/adapter.rb

```diff
@@ -14,9 +16,67 @@ def self.create(resource, options = {})
         klass.new(resource, options)
       end
 
+      # @see ActiveModel::Serializer::Adapter.get
       def self.adapter_class(adapter)
-        adapter_name = adapter.to_s.classify.sub('API', 'Api')
-        "ActiveModel::Serializer::Adapter::#{adapter_name}".safe_constantize
+        ActiveModel::Serializer::Adapter.get(adapter)
+      end
+
+      # Only the Adapter class has these methods.
+      # None of the sublasses have them.
+      class << ActiveModel::Serializer::Adapter
+        # @return Hash<adapter_name, adapter_class>
+        def adapter_map
+          ADAPTER_MAP
+        end
+
+        # @return Array<adapter_name>
+        def adapters
+          adapter_map.keys.sort
+        end
+
+        # Adds an adapter 'klass' with 'name' to the 'adapter_map'
+        # Names are stringified and underscored
+        def register(name, klass)
+          adapter_map.update(name.to_s.underscore => klass)
+          self
+        end
+
+        # @param  adapter [String, Symbol, Class] name to fetch adapter by
+        # @return [ActiveModel::Serializer::Adapter] subclass of Adapter
+        # @raise  [UnknownAdapterError]
+        def get(adapter)
+          # 1. return if is a class
```

Comment:
@sandstrom re: ActiveJob [QueueAdapter](https://github.com/rails/rails/blob/a11571cec3213753d63ac3e6b4bb3b97fe2594a6/activejob/lib/active_job/queue_adapter.rb#L27-L60) and [QueueAdapters](https://github.com/rails/rails/blob/a11571cec3213753d63ac3e6b4bb3b97fe2594a6/activejob/lib/active_job/queue_adapters.rb#L123-L133) I think I like `lookup` more than `get`, so thanks for pointing me that way ;)  You think I should change it?

However, I think there are some limitations in that interface in that 

1. activejob queue adapters are only used when assigned via `queue_adapter =`, but there's no way to know if and when an AMS adapter will be used
2. I like `const_get` instead of `safe_constantize`, but we'd still need to rescue a NameError and use the AM::S::Adapter namespace, at least for dynamic lookup
3. I like that you can pass [`interpret_adapter` a class and it can check it for behavior](https://github.com/rails/rails/blob/a11571cec3213753d63ac3e6b4bb3b97fe2594a6/activejob/lib/active_job/queue_adapter.rb#L52-L56), but I think that's outside the scope of this PR
4. I'm glad to get the reminder about private_constant